model/common/src/icon4py/model/common/dimension.py

```diff
@@ -6,53 +6,64 @@
+MAIN_HORIZONTAL_DIMENSIONS = {"CellDim": CellDim, "EdgeDim": EdgeDim, "VertexDim": VertexDim}
+CEDim = gtx.Dimension("CE")
+CECDim = gtx.Dimension("CEC")
+ECDim = gtx.Dimension("EC")
+ECVDim = gtx.Dimension("ECV")
+CECECDim = gtx.Dimension("CECEC")
+E2CDim = gtx.Dimension("E2C", gtx.DimensionKind.LOCAL)
+E2VDim = gtx.Dimension("E2V", gtx.DimensionKind.LOCAL)
+C2EDim = gtx.Dimension("C2E", gtx.DimensionKind.LOCAL)
+V2CDim = gtx.Dimension("V2C", gtx.DimensionKind.LOCAL)
+C2VDim = gtx.Dimension("C2V", gtx.DimensionKind.LOCAL)
+V2EDim = gtx.Dimension("V2E", gtx.DimensionKind.LOCAL)
+V2E2VDim = gtx.Dimension("V2E2V", gtx.DimensionKind.LOCAL)
+E2C2VDim = gtx.Dimension("E2C2V", gtx.DimensionKind.LOCAL)
+C2E2CODim = gtx.Dimension("C2E2CO", gtx.DimensionKind.LOCAL)
+E2C2EODim = gtx.Dimension("E2C2EO", gtx.DimensionKind.LOCAL)
+E2C2EDim = gtx.Dimension("E2C2E", gtx.DimensionKind.LOCAL)
+C2E2CDim = gtx.Dimension("C2E2C", gtx.DimensionKind.LOCAL)
+C2E2C2EDim = gtx.Dimension("C2E2C2E", gtx.DimensionKind.LOCAL)
+C2E2C2E2CDim = gtx.Dimension("C2E2C2E2C", gtx.DimensionKind.LOCAL)
+E2C = gtx.FieldOffset("E2C", source=CellDim, target=(EdgeDim, E2CDim))
+C2E = gtx.FieldOffset("C2E", source=EdgeDim, target=(CellDim, C2EDim))
+V2C = gtx.FieldOffset("V2C", source=CellDim, target=(VertexDim, V2CDim))
+C2V = gtx.FieldOffset("C2V", source=VertexDim, target=(CellDim, C2VDim))
+V2E = gtx.FieldOffset("V2E", source=EdgeDim, target=(VertexDim, V2EDim))
+E2V = gtx.FieldOffset("E2V", source=VertexDim, target=(EdgeDim, E2VDim))
+C2CE = gtx.FieldOffset("C2CE", source=CEDim, target=(CellDim, C2EDim))
+C2CEC = gtx.FieldOffset("C2CEC", source=CECDim, target=(CellDim, C2E2CDim))
+E2EC = gtx.FieldOffset("E2EC", source=ECDim, target=(EdgeDim, E2CDim))
+E2ECV = gtx.FieldOffset("E2ECV", source=ECVDim, target=(EdgeDim, E2C2VDim))
+E2C2V = gtx.FieldOffset("E2C2V", source=VertexDim, target=(EdgeDim, E2C2VDim))
+C2E2CO = gtx.FieldOffset("C2E2CO", source=CellDim, target=(CellDim, C2E2CODim))
+E2C2EO = gtx.FieldOffset("E2C2EO", source=EdgeDim, target=(EdgeDim, E2C2EODim))
+E2C2E = gtx.FieldOffset("E2C2E", source=EdgeDim, target=(EdgeDim, E2C2EDim))
+C2E2C = gtx.FieldOffset("C2E2C", source=CellDim, target=(CellDim, C2E2CDim))
+C2E2C2E = gtx.FieldOffset("C2E2C2E", source=EdgeDim, target=(CellDim, C2E2C2EDim))
+C2E2C2E2C = gtx.FieldOffset("C2E2C2E2C", source=CellDim, target=(CellDim, C2E2C2E2CDim))
+C2CECEC = gtx.FieldOffset("C2CECEC", source=CECECDim, target=(CellDim, C2E2C2E2CDim))
+V2E2V = gtx.FieldOffset("V2E2V", source=VertexDim, target=(VertexDim, V2E2VDim))
+Koff = gtx.FieldOffset("Koff", source=KDim, target=(KDim,))
+KHalfOff = gtx.FieldOffset("KHalfOff", source=KHalfDim, target=(KHalfDim,))
+
+
+def from_value(name: str) -> gtx.Dimension | None:
+    """Get the dimension from its name."""
+    return next(
+        (
+            value
+            for value in globals().values()
+            if isinstance(value, gtx.Dimension) and value.value == name
+        ),
+        None,
+    )
```

Comment:
Maybe it's simpler to build a static dictionary from offset name or offset instance to dimension, and then access it when needed.